sklearnex/linear_model/tests/test_linear.py

```diff
@@ -52,9 +52,9 @@ def test_sklearnex_import_linear(dataframe, queue, dtype, macro_block):
     assert "sklearnex" in linreg.__module__
     assert linreg.n_features_in_ == 2
 
-    tol = 1e-5 if X.dtype == np.float32 else 1e-7
+    tol = 1e-5 if dtype == np.float32 else 1e-7

```

Comment:
Btw you probably can estimate the upper bound of imperfection for calculations as 'np.finfo(dtype).eps * n_operations', where 'n_operations' is intrinsic to an algorithm and generally should be equal to 'n_cols' in case of LinearRegression, PCA & Covariance; 'n_rows' for knn, kmeans

Reply:
@olegkkruglov what do you think?